### DISCUSSION TITLE
About model saved files after training

### DISCUSSION THREAD

**Discussion by nihil39** [ACEsuit/mace#613] [Q&A] [answered]
Hi, 

after the model training is finished four files are saved (example below). 

What's the difference between compiled and not compiled models?

The swa suffix means after swa has started? What is the file to be used in ASE or LAMMPS to evaluate the model? 

Thanks

```
MACE_model_Rev_PBE_D3_1000_confs_L0_128_hi_E0s_fw_1_ew_1_epochs_200_loss_stress_bs_10_lr_001_compiled.model
MACE_model_Rev_PBE_D3_1000_confs_L0_128_hi_E0s_fw_1_ew_1_epochs_200_loss_stress_bs_10_lr_001.model
MACE_model_Rev_PBE_D3_1000_confs_L0_128_hi_E0s_fw_1_ew_1_epochs_200_loss_stress_bs_10_lr_001_swa_compiled.model
MACE_model_Rev_PBE_D3_1000_confs_L0_128_hi_E0s_fw_1_ew_1_epochs_200_loss_stress_bs_10_lr_001_swa.model
```

**Comment by ilyes319** [comment, marked as answer]
The compiled model can be exported to C++ for example for OpenMM and contains some metadata on the model training. The non compiled are to be used in python. The swa is the best model from the swa phase (renamed second_phase) and the one without swa is the best model before that phase.
